### PR TITLE
Allow configuring TLS profiles for our externally facing components

### PR DESCRIPTION
**Pull request by akalenyu**
**What this PR does / why we need it**:
TLS security profiles provide a way for servers to regulate which ciphers/minimum TLS version a client can use when connecting to the server.
Security-minded cluster administrators might want to set specific (even custom) TLS profiles for our externally-facing components,
such as the uploadproxy and the apiserver; let's give them the option to do that.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
Cluster administrators can set TLS profiles for CDI's externally-facing components
```

### CODE REVIEW

File: pkg/util/tls-crypto-watch/tls-crypto-watch.go [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+	newConfig.CipherSuites = ciphers
+	newConfig.MinVersion = minTLSVersion
+
+	ctw.mutex.Lock()
+	defer ctw.mutex.Unlock()
+	ctw.config = newConfig
+}
+
+// SelectCipherSuitesAndMinTLSVersion returns cipher names and minimal TLS version according to the input profile
+func SelectCipherSuitesAndMinTLSVersion(profile *ocpconfigv1.TLSSecurityProfile) ([]string, ocpconfigv1.TLSProtocolVersion) {
```
Comment (mhenriks) [inline pkg/util/tls-crypto-watch/tls-crypto-watch.go:125]: My big question with this PR, and it's really probably not a big deal, is would it be better to return some defaults here, or defer to whatever the default golang is using.  Worst case theoretically, a cipher is compromised and we don't know about it.  Go library would stop using the cipher when we update the compiler.

**Review by mhenriks — COMMENTED** [review]
I think this looks really great just need to update the api package go.mod

File: staging/src/kubevirt.io/containerized-data-importer-api/go.mod [hunk 2]
```diff
@@ -26,3 +27,5 @@ require (
 	sigs.k8s.io/json v0.0.0-20211020170558-c049b76a60c6 // indirect
 	sigs.k8s.io/structured-merge-diff/v4 v4.2.1 // indirect
 )
+
+replace github.com/openshift/api => github.com/openshift/api v0.0.0-20211217221424-8779abfbd571
```
[the remark this answers is not in the document]
Comment (akalenyu) [reply staging/src/kubevirt.io/containerized-data-importer-api/go.mod:31]: gotcha
So I am trying to understand how versioning works for this and this is confusing:
https://pkg.go.dev/github.com/openshift/api/config/v1?tab=versions

Also, would this mean I'd have to bump the k8s packages here to 1.24 to match the openshift one?

File: pkg/util/tls-crypto-watch/tls-crypto-watch.go (hunk 1 above)
Comment (akalenyu) [reply pkg/util/tls-crypto-watch/tls-crypto-watch.go:125]: So these defaults are based on https://wiki.mozilla.org/Security/Server_Side_TLS#Intermediate_compatibility_.28recommended.29
Which is advised as the "recommended" for the vast majority of services.

But I also wouldn't mind omitting and deferring to go. Just want to make sure that is what we want

File: staging/src/kubevirt.io/containerized-data-importer-api/go.mod (hunk 2 above)
[the remark this answers is not in the document]
Comment (mhenriks) [reply staging/src/kubevirt.io/containerized-data-importer-api/go.mod:31]: For now, I would just get rid of the replace line and use that version in the require stanza

[the remark this answers is not in the document]
Comment (akalenyu) [reply staging/src/kubevirt.io/containerized-data-importer-api/go.mod]: Done

**Pull request merged**